File: sorting-algorithms/selection-sort/JS/selection-sort.js
function selectionSort(arr){

    const size = arr.length;
    const sorted = [...arr];
    
    for(let i = 0; i < size - 1; ++i){
        let minIndex = i;
        for(let j = i + 1; j < size; ++j){
            if(sorted[j] < sorted[minIndex]){
                minIndex = j;
            }
        }
        if(minIndex != i){
            [sorted[minIndex],sorted[i]] = [sorted[i],sorted[minIndex]];
        }
    }

    return sorted;

}

let arr = [64,25,12,22,11];

selectionSort(arr);
console.log(arr);

File: algorithms/sorting-algorithms/selection-sort/C/selection-sort.c
#include <stdio.h>

void selectionSort(int[],int size);

int main(){

    int arr[5] = {64,25,12,22,11};

    selectionSort(arr,5);

    for(int i = 0; i < 5; ++i)
    {
        printf("%d ",arr[i]);
    }

    return 0;

}


void selectionSort(int arr[],int size){
    for(int i = 0; i < size - 1; ++i)
    {
        int minIndex = i;
        for(int j = i + 1; j < size; ++j)
        {
            if(arr[j] < arr[minIndex])
            {
                minIndex = j;
            }
        }
        if(i != minIndex)
        {
            int tmp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = tmp;
        }
    }
}